From 3b8957aad94077286527d45b8827481736fa919c Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Tue, 13 Feb 2018 14:24:38 +1100
Subject: [PATCH] digital-clock: enable RFC2822 date format

This allows a nicer clock format to use as the Solus default.

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 applets/digital-clock/package/contents/config/main.xml         | 2 +-
 applets/digital-clock/package/contents/ui/DigitalClock.qml     | 2 ++
 applets/digital-clock/package/contents/ui/configAppearance.qml | 4 ++++
 3 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/applets/digital-clock/package/contents/config/main.xml b/applets/digital-clock/package/contents/config/main.xml
index df89e0c..5db7f0e 100644
--- a/applets/digital-clock/package/contents/config/main.xml
+++ b/applets/digital-clock/package/contents/config/main.xml
@@ -19,7 +19,7 @@
       <default>false</default>
     </entry>
     <entry name="dateFormat" type="string">
-      <label>The date format to display. Options are: shortDate, longDate or isoDate.</label>
+      <label>The date format to display. Options are: shortDate, longDate, isoDate or rfc2822Date.</label>
       <default>shortDate</default>
     </entry>
     <entry name="fontFamily" type="string">
diff --git a/applets/digital-clock/package/contents/ui/DigitalClock.qml b/applets/digital-clock/package/contents/ui/DigitalClock.qml
index 1362fe7..df3dc2e 100644
--- a/applets/digital-clock/package/contents/ui/DigitalClock.qml
+++ b/applets/digital-clock/package/contents/ui/DigitalClock.qml
@@ -38,6 +38,8 @@ Item {
             return  Qt.SystemLocaleLongDate;
         } else if (plasmoid.configuration.dateFormat === "isoDate") {
             return Qt.ISODate;
+        } else if (plasmoid.configuration.dateFormat === "rfc2822Date") {
+            return Qt.RFC2822Date;
         }
 
         return Qt.SystemLocaleShortDate;
diff --git a/applets/digital-clock/package/contents/ui/configAppearance.qml b/applets/digital-clock/package/contents/ui/configAppearance.qml
index 83d95bf..96d8002 100644
--- a/applets/digital-clock/package/contents/ui/configAppearance.qml
+++ b/applets/digital-clock/package/contents/ui/configAppearance.qml
@@ -147,6 +147,10 @@ Item {
                             {
                                 'label': i18n("ISO Date"),
                                 'name': "isoDate"
+                            },
+                            {
+                                'label': i18n("RFC2822 Date"),
+                                'name': "rfc2822Date"
                             }
                         ]
                         onCurrentIndexChanged: cfg_dateFormat = model[currentIndex]["name"]
-- 
2.16.1

